src/resmom/linux/mom_start.c

```diff
@@ -2436,11 +2436,13 @@ scan_for_terminated(void)
 
 	termin_child = 0;
 
-	if (mom_get_sample() == PBSE_NONE) {
-		pjob = (job *)GET_NEXT(svr_alljobs);
-		while (pjob) {
-			mom_set_use(pjob);
-			pjob = (job *)GET_NEXT(pjob->ji_alljobs);
+	if (!mock_run) {
+		if (mom_get_sample() == PBSE_NONE) {
```

Comment:
there are other places also mom_get_sample and traversing of svr_alljobs is used, so shouldn't we update those?

Reply:
i hadn't seen mom_get_sample() crop up on perf profiling after I added it here, so I didn't look further into it, thanks for pointing it out, I'll add a check at other places as well.